src/OrchardCore.Modules/OrchardCore.Users/Controllers/ControllerExtensions.cs

```diff
@@ -0,0 +1,108 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using OrchardCore.DisplayManagement;
+using OrchardCore.Email;
+using OrchardCore.Entities;
+using OrchardCore.Modules;
+using OrchardCore.Settings;
+using OrchardCore.Users.Events;
+using OrchardCore.Users.Models;
+using OrchardCore.Users.Services;
+using OrchardCore.Users.ViewModels;
+
+namespace OrchardCore.Users.Controllers
+{
+    internal static class ControllerExtensions
```

Comment:
I dislike this one, why we don't use `ISmtpService` directly to send emails, same thing for others

Reply:
There was a base controller that account and registration inherited from that one, only to reuse the email functionality. In order to be reusable and remove the base controller, i created that extension class.

Reply:
It's fine while it's `internal` :)